conda-store-server/conda_store_server/api.py

```diff
@@ -58,6 +57,20 @@ def update_namespace(
     if metadata_ is not None:
         namespace.metadata_ = metadata_
 
+    db.commit()
+
+    return namespace
+
+
+def update_namespace_role(
+    db,
+    name: str,
+    role_mappings: Dict[str, List[str]] = None,
```

Comment:
This is interesting. Because this is typed `List[str]` instead of `Any`, like metadata, this now gets rejected by the FastAPI router with 422. In order to fix it, I had to make this change to the route signature:

```py
    role_mappings: Dict[str, List[str]] = None,  # doesn't work
    role_mappings: Dict[str, Any] = None,  # works
```

Printing `request.body()` before the route is being processed, I see the same value in both cases. So it must be an internal issue related to serialization, like this one: https://github.com/tiangolo/fastapi/issues/3373

Code to print request body (add it to `conda-store-server/conda_store_server/server/app.py`):

```py
        if self.enable_api:
            from fastapi import Depends
            async def log_json(request: Request):
                print("XXXXXXXXXXXXXXXXX", await request.body())
            app.include_router(
                views.router_api,
                prefix=trim_slash(self.url_prefix),
                dependencies=[Depends(log_json)],
            )
```

Reply:
Ah, I see. After splitting the route into two, it's no longer necessary to include the keyword matching the route parameter. So no need to pass `role_mappings: dict`. This can be just the inner dict directly.